flink-streaming-java/src/main/java/org/apache/flink/streaming/api/datastream/WindowedStream.java

```diff
@@ -459,7 +545,28 @@ public WindowedStream(KeyedStream<T, K> input,
 	 * @param resultType Type information for the result type of the window function
 	 * @return The data stream that is the result of applying the window function to the window.
 	 */
-	public <R> SingleOutputStreamOperator<R> apply(R initialValue, FoldFunction<T, R> foldFunction, WindowFunction<R, R, K, W> function, TypeInformation<R> resultType) {
+	public <R> SingleOutputStreamOperator<R> apply(R initialValue, FoldFunction<T, R> foldFunction,
```

Comment:
Thx for the review, first of all. I see your point here. I will introduce these ProcessWindowFunction-specific fold/reduce methods, w/o altering the old WindowFunction API of course.